script.js

```diff
@@ -0,0 +1,365 @@
+const API_URL = 'https://raw.githubusercontent.com/pahanan/pahan_an.github.io/refs/heads/main/api/data.json';
+const INGREDIENTS_API_URL = 'https://raw.githubusercontent.com/pahanan/pahan_an.github.io/refs/heads/main/api/ingredient-prices.json';
+let recipes = [];
+let ingredientPrices = {};
+
+const modal = document.querySelector('#modal');
+const closeModalButton = document.querySelector('#close-modal');
+const addNewRecipeButton = document.querySelector('#add-new-recipe');
+const ingredientsContainer = document.querySelector('#ingredients-container');
+const recipeForm = document.querySelector('#recipe-form');
+const recipesContainer = document.querySelector('#recipes-container');
+const searchInput = document.querySelector('#word-for-search');
+const ingredientSearchInput = document.querySelector('#ingredient-for-search');
+const searchButton = document.querySelector('#search-button');
+const searchingredientButton = document.querySelector('#ingredient-search-button');
+
+
+async function fetchRecipes() {
```

Comment:
The function has too many responsibilities. A function should only have one job/responsibility. 
According to the name `fetchRecipes` should return recipes.
Down in line 364 you can call all of your functions 
can be something like 
```
const recipes = await fetchRecipes();
const ingredientPrices = await fetchIngredientPrices();
enrichRecipesWithPrices(recipes, ingredientPrices)
renderRecipes(recipes)
```